File: JS/programmers_의상.js
function solution(clothes) {
    const map = new Map();
    for (const cloth of clothes) {
        const [name, sort] = cloth;
        if(map.has(sort)) map.set(sort, map.get(sort) + 1);
        else map.set(sort, 1);
    }
    console.log(Array.from(map.values()))
    
    return Array.from(map.values()).reduce((acc, cur) => acc * (cur + 1), 1) - 1;
}

File: Python/programmers_의상.py
def solution(clothes):
    answer = 1
    dic = {}
    for cloth in clothes:
        name, sort = cloth
        if sort in dic:
            dic[sort] = dic[sort] + 1
        else:
            dic[sort] = 1
    print(dic)
    for key, value in dic.items():
        answer = answer * (value + 1)
        
    return answer - 1